ssl/ssl_ciph.c

```diff
@@ -2221,8 +2231,13 @@ int ssl_cipher_get_overhead(const SSL_CIPHER *c, size_t *mac_overhead,
 
 int ssl_cert_is_disabled(SSL_CTX *ctx, size_t idx)
 {
-    const SSL_CERT_LOOKUP *cl = ssl_cert_lookup_by_idx(idx);
+    SSL_CERT_LOOKUP *cl;
+
+    /* A provider-loaded key type is always enabled */
+    if (idx >= SSL_PKEY_NUM)
```

Comment:
Sorry, don't get logic here. Could you please elaborate?

It may work this way now, but we may want to exclude several of provided algorithms, I think.

Reply:
> Sorry, don't get logic here. Could you please elaborate?
> 
> It may work this way now, but we may want to exclude several of provided algorithms, I think.

The thinking was that if you don't want a provider's algorithms, you don't enable it. Or in reverse: If you enabled the provider, you want it's algorithms. Thus they're active.

If a provider is too "coarse grained", i.e., that it carries so many algorithms that you want to en/disable single algorithms _within_ the provider, this would need to be reconsidered. In such case, though, I wonder how this should be facilitated: Would that be a provider-specific mechanism? If so, the provider thus configured simply would not report disabled algorithms to OpenSSL, i.e., they never would become visible (e.g., in the test above). If no, i.e., you have a generic <, provider-independent mechanism in mind, please let me know about that. That then should give us an idea how to change the test.

Reply:
Currently we have cipherstring mapping to permitted and forbidden masks, and I think that this mechanism should continue working fir provider-based algorithms

Reply:
I'm not sure I understand how cipherstrings and signature algorithms relate: Could you please point to the mapping you refer to (and ideally, how those masks should work with provided sigalgs)? Thanks in advance.

Reply:
I mean the `SSL_CTX_set_cipher_list` and its descendant calls. Am I mistaken that signature algorithm filter is applicable there?

Reply:
In TLSv1.2 and below sigalgs and ciphersuites are inextricably linked. But from TLSv1.3 they are entirely separate. Since the pluggable sigalgs support only applies to TLSv1.3 I don't think there is any link to the `SSL_CTX_set_cipher_list` functionality.

Reply:
Thanks for the insight, @mattcaswell . OK to consider the topic resolved then, @beldmit ?

Reply:
I think yes, thank you!